### DISCUSSION TITLE
[2021-05-20] Amplify Office Hours Question Queue

### DISCUSSION THREAD

**Discussion by medelman17** [aws-amplify/learn#328] [Office Hours]
Howdy, folks! Have a question for the Amplify team? Post it here and we'll try to address it during our upcoming office hours! 

We'll be going live on Discord at 12 PM PDT / 3 PM EDT on Thursday, May 20, 2021. Join here! Link: https://go.aws/3fwIWOP

**Comment by estyh** [comment]
I get very confused about the line between amplify backend environments and frontend - which are branch based, but also connected to a specific environment?? So, if I have three environments, because I am testing 3 features, or sandbox environments for different devs, I need to hook up each branch to each environment? Also, something I can't find answer to anywhere, in the amplify.yml buildspec for monorepos, are the baseDirectory and cache paths relative to the appRoot or to the monorepo root?
Thanks!

**Reply by estyh** [reply]
oh sorry, don't have mic connected!!

**Reply by siegerts** [reply]
Np!

**Reply by estyh** [reply]
got it on now

**Reply by medelman17** [reply]
Note: We should add clarity to docs re above processes generally.

**Reply by estyh** [reply]
Yup, a team-oriented ci/cd guide would be great! Specifically, creating new ad-hoc environments (not the standard dev/staging/prod), auto-linking them to existing frontend app (apps, for a monorepo). And, what you guys brought up, about seeding the new environments with data, super important.
Thanks!

**Comment by mo-malaka** [comment]
[AWS Whitepaper] Amplify DataStore: Use Cases and Implementation https://docs.aws.amazon.com/whitepapers/latest/amplify-datastore-implementation/introduction.html

**Comment by byblakeorriver** [comment]
I am trying to use the generated graphql subscriptions in a server side environment (node). I keep getting an error that says `Connection failed: WebSocket is not defined"`. Can you let me know how to define this websocket?

**Reply by byblakeorriver** [reply]
Here is the entire error: 
```
"error:  {"provider":{"_config":{"aws_project_region":"us-west-2","aws_appsync_graphqlEndpoint":"https://xxxxxxxxx.appsync-api.us-west-2.amazonaws.com/graphql/%22,/%22aws_appsync_region/%22:/%22us-west-2/%22,/%22aws_appsync_authenticationType/%22:/%22API_KEY/%22,/%22aws_appsync_apiKey/%22:/%22XXXXXXXXXXX/%22%7D,/%22socketStatus/%22:0,/%22keepAliveTimeout/%22:300000,/%22subscriptionObserverMap/%22:%7B%7D,/%22promiseArray/%22:[],/%22awsRealTimeSocket/%22:null%7D,/%22error/%22:%7B/%22errors/%22:[%7B/%22message/%22:/%22Connection failed: WebSocket is not defined"}]}}"
```

**Reply by byblakeorriver** [reply]
Here is my Amplify config: 
```
import {Amplify} from "aws-amplify";

Amplify.configure({
  aws_project_region: "us-west-2",
  aws_appsync_graphqlEndpoint: "https://XXXXX.appsync-api.us-west-2.amazonaws.com/graphql",
  aws_appsync_region: "us-west-2",
  aws_appsync_authenticationType: "API_KEY",
  aws_appsync_apiKey: "XXXXX",
});
```

**Reply by danshirley1** [reply]
Did you resolve this?

**Reply by byblakeorriver** [reply]
@danshirley1 I did not resolve this. I eventually came to the understanding that you cannot use the client server side. I could have been wrong though.

**Comment by jobypthomas** [comment]
I have a question - about the stack limit - https://github.com/aws-amplify/amplify-cli/issues/2467 The limit is fairly low. Any updates?

**Comment by MiladNazeri** [comment]
I have two types that look like this(stripped down):
type Show @model
@key(name: "byHost", fields: ["hostID", "showtime"], queryField: "showsByHost")
{
    hostID: ID! 
    host: Host! @connection(fields: ["hostID"]) 
}

type Host @model
{
    id: ID! 
    shows: [Show] @connection(keyName: "byHost", fields: ["id"])
}

My question is that hostID must be required I believe under Shows.  But in the case where a show doesn't have a host, how can I handle this?  Is there a way to do an optional connection?
